include/zephyr/devicetree.h

```diff
@@ -982,6 +984,67 @@
 	COND_CODE_1(DT_NODE_HAS_PROP(node_id, prop), \
 		(DT_STRING_UPPER_TOKEN(node_id, prop)), (default_value))
 
+/**
+ * @brief Get a string property's value as an unquoted string content.
+ *
+ * This removes "the quotes" from string-valued properties.
+ * That can be useful, for example,
+ * when defining floating point values as a string or a equation
+ * that may contain functions to call.
```

Comment:
```suggestion
 * @brief Get a string property's value as an unquoted sequence of tokens
 *
 * This removes "the quotes" from string-valued properties.
 * That can be useful, for example,
 * when defining floating point values as a string in devicetree
 * that you would like to use to initialize a float or double variable in C.
```

The "functions to call" thing is powerful but scares the heck out of me; I'd rather keep that out of this official docstring ;)

Reply:
I got you. I thought that from embedded world perspective, having access to the DTS is basically the same like having access to source code. You have to recompile it anyway to have new DTS included. But I think that any advanced user would figure it out by itself and we do not have to inform about this functionality officially :)